### PR TITLE
add option to use event zoroark to tournament farmer

### PR DESCRIPTION
**Pull request by kichithewolf**

### CODE REVIEW

**Comment by kichithewolf** [comment]
Regex:
image

File: SerialPrograms/Source/PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h [hunk 1]
```diff
@@ -88,6 +88,27 @@ class TerastallizingDetector : public StaticScreenDetector{
...
 
 
+class SwapMenuDetector : public StaticScreenDetector {
+public:
+    SwapMenuDetector(Color color);
+
+    virtual void make_overlays(VideoOverlaySet& items) const override;
+    virtual bool detect(const ImageViewRGB32& screen) const override;
+
+    //  Returns -1 if not found.
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h:98]: If that commented code was taken from somewhere else, it should be removed!

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp [hunk 2]
```diff
@@ -220,21 +332,21 @@ void TournamentFarmer::run_battle(SingleSwitchProgramEnvironment& env, BotBaseCo
                 NormalBattleMenuWatcher battle_menu(COLOR_YELLOW);
                 int ret = wait_until(
                     env.console, context,
-                    std::chrono::seconds(60), //Tera takes ~25 to 30 seconds for player/opponent
+                    std::chrono::seconds(90), //Tera takes ~25 to 30 seconds for player/opponent
                     { battle_menu } //End of battle from tera'd ace takes longer, 45 seconds was not enough
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp:336]: Comment should be updated to 90s ?

**Review by pifopi — APPROVED** [review]
Looks good to me but I left some comments!

File: SerialPrograms/Source/PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h (hunk 1 above)
Comment (kichithewolf) [reply SerialPrograms/Source/PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h]: Removed. I was originally going to implement them as well but realized it wasn't needed for this program.

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp (hunk 2 above)
Comment (kichithewolf) [reply SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp:336]: Updated comment for clarity.

**Pull request merged**